extensions/extension-rest-api-call/src/main/java/org/thingsboard/server/extensions/rest/action/DeviceAttributeRestAction.java

```diff
@@ -0,0 +1,101 @@
+package org.thingsboard.server.extensions.rest.action;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.thingsboard.server.common.data.kv.AttributeKvEntry;
+import org.thingsboard.server.common.msg.core.BasicUpdateAttributesRequest;
+import org.thingsboard.server.common.msg.device.ToDeviceActorMsg;
+import org.thingsboard.server.common.msg.session.ToDeviceMsg;
+import org.thingsboard.server.extensions.api.component.Action;
+import org.thingsboard.server.extensions.api.plugins.PluginAction;
+import org.thingsboard.server.extensions.api.plugins.msg.PluginToRuleMsg;
+import org.thingsboard.server.extensions.api.plugins.msg.RuleToPluginMsg;
+import org.thingsboard.server.extensions.api.rules.RuleContext;
+import org.thingsboard.server.extensions.api.rules.RuleProcessingMetaData;
+
+import java.util.*;
+import java.util.logging.Logger;
+
+@Action(name = "Device Attributes Rest Plugin Action",
+        descriptor = "DeviceAttributeRestDescriptor.json", configuration = RestApiCallPluginActionConfiguration.class)
+@Slf4j
+public class DeviceAttributeRestAction implements PluginAction<RestApiCallPluginActionConfiguration>{
+    private RestApiCallPluginActionConfiguration configuration;
+    private final org.slf4j.Logger logger = LoggerFactory.getLogger(DeviceAttributeRestAction.class);
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public void suspend() {
+
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public void init(RestApiCallPluginActionConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Optional<RuleToPluginMsg<?>> convert(RuleContext ctx, ToDeviceActorMsg msg, RuleProcessingMetaData deviceMsgMd) {
+        BasicUpdateAttributesRequest payload;
+
+        String device = ctx.getDeviceMetaData().getDeviceName();
+        if (msg.getPayload() instanceof BasicUpdateAttributesRequest) {
+            payload = (BasicUpdateAttributesRequest) msg.getPayload();
+        } else {
+            throw new IllegalArgumentException("Action does not support messages of type: " + msg.getPayload().getMsgType());
+        }
+
+        HashMap<String, String> attributes = new HashMap<>(payload.getAttributes().size());
+        attributes.put("deviceName",device);
+        String tagsStr = "";
+        //Comma separated tags.
+        for(AttributeKvEntry attr: payload.getAttributes()){
+            tagsStr = tagsStr + attr.getValueAsString() + ",";
+            //attributes.put(attr.getKey(), attr.getValueAsString());
+        }
```

Comment:
As a good practice, never use String concatenation to build a String from Collection as it's not performant. Use StringBuilder. But you could find Utility methods in StringUtils to join list into string with separator. Check Java 8 Streams as well.